system/known_hosts.py

```diff
@@ -0,0 +1,234 @@
+#!/usr/bin/env python
+
+"""
+Ansible module to manage the ssh known_hosts file.
+Copyright(c) 2014, Matthew Vernon <mcv21@cam.ac.uk>
+
+This module is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This module is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this module.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+DOCUMENTATION = '''
+---
+module: known_hosts
+short_description: Add or remove a host from the C(known_hosts) file
+description:
+   - The M(known_hosts) module lets you add or remove a host from the C(known_hosts) file. 
+     This is useful if you're going to want to use the M(git) module over ssh, for example. 
+     If you have a very large number of host keys to manage, you will find the M(template) module more useful.
+version_added: "1.6"
+options:
+  host:
+    description:
+      - The host to add or remove (must match a host specified in key)
+    required: true
+    default: null
+  key:
+    description:
+      - The SSH public host key, as a string (optional if removing the host)
```

Comment:
change to: required when state=present, optional when state=absent in which case all keys for a host will be removed